File: scripts/patch-rnmapbox.py
#!/usr/bin/env python3
import re

file_path = "node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/styles/RNMBXStyleFactory.kt"

# Read the file
with open(file_path, 'r') as f:
    content = f.read()

# Store original content for backup
original_content = content

# List of deprecated functions to comment out
deprecated_functions = [
    'setFillPatternCrossFade',
    'setLineElevationReference',
    'setLineCrossSlope',
    'setLinePatternCrossFade',
    'setCircleElevationReference',
    'setFillExtrusionPatternCrossFade',
    'setFillExtrusionHeightAlignment',
    'setFillExtrusionBaseAlignment',
    'setBackgroundPitchAlignment'
]

# For each function, find and comment out
for func_name in deprecated_functions:
    # Pattern to match function definition and its body
    pattern = rf'(\s+)(fun {func_name}\([^{{]+\{{(?:[^{{}}]*\{{[^{{}}]*\}})*[^{{}}]*\}})'
    
    def replace_func(match):
        indent = match.group(1)
        func_body = match.group(2)
        # Comment out each line
        commented = '\n'.join([indent + '// ' + line.lstrip() if line.strip() else line 
                               for line in func_body.split('\n')])
        return commented
    
    content = re.sub(pattern, replace_func, content, flags=re.MULTILINE | re.DOTALL)

# Save backup of original content
with open(file_path + '.bak', 'w') as f:
    f.write(original_content)

# Write patched content
with open(file_path, 'w') as f:
    f.write(content)

print(f"Successfully patched {file_path}")
print("Backup saved as {}.bak".format(file_path))

File: scripts/patch-rnmapbox.pl
#!/usr/bin/env perl
use strict;
use warnings;

my $file = 'node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/styles/RNMBXStyleFactory.kt';

# Read file
open(my $fh, '<', $file) or die "Cannot open $file: $!";
my @lines = <$fh>;
close($fh);

# Backup
open(my $bak, '>', "$file.backup") or die "Cannot create backup: $!";
print $bak @lines;
close($bak);

my @functions = qw(
    setFillPatternCrossFade
    setLineElevationReference
    setLineCrossSlope
    setLinePatternCrossFade
    setCircleElevationReference
    setFillExtrusionPatternCrossFade
    setFillExtrusionHeightAlignment
    setFillExtrusionBaseAlignment
    setBackgroundPitchAlignment
);

my $in_function = 0;
my $brace_count = 0;
my @output;

foreach my $line (@lines) {
    # Check if we're starting a function we want to comment
    if (!$in_function) {
        foreach my $func (@functions) {
            if ($line =~ /fun\s+$func\s*\(/) {
                $in_function = 1;
                $brace_count = 0;
                last;
            }
        }
    }
    
    if ($in_function) {
        # Count braces
        my $open = () = $line =~ /{/g;
        my $close = () = $line =~ /}/g;
        $brace_count += $open - $close;
        
        # Comment the line
        if ($line =~ /^(\s*)(.+)$/) {
            push @output, "$1// $2\n";
        } else {
            push @output, $line;
        }
        
        # Check if function is complete - when we have at least one opening brace and braces are balanced
        if ($brace_count == 0 && $close > 0) {
            $in_function = 0;
        }
    } else {
        push @output, $line;
    }
}

# Write output
open(my $out, '>', $file) or die "Cannot write to $file: $!";
print $out @output;
close($out);

print "Successfully patched $file\n";
print "Backup saved to $file.backup\n";
